packages/pixeloven/cli-addon-generators/src/templates/app-ssr/src/server/middleware/renderer.tsx

```diff
@@ -0,0 +1,150 @@
+                scripts={props.req.files && <Script src={props.req.files.js} />}
+                state={props.store.getState()}
+            >
+                {content}
+            </Body>
+        </Html>,
+    );
+    return `<!DOCTYPE html>${template}`;
+}
+
+export function errorHandler(config: Config) {
+    return (req: Request, res: Response, next: NextFunction) => {
+        const staticContext: StaticContext = {
+            statusCode: 500,
+        };
+        try {
+            const store = configureStore("server");
+            const routes = Routing.getConfig(
+                unknownErrorRoutes,
+                config.publicPath,
+            );
+            store
+                .runSaga(rootSaga)
+                .toPromise()
+                .then(() => {
+                    const output = Template({
+                        req,
+                        routes,
+                        staticContext,
+                        store,
+                    });
+                    const statusCode = staticContext.statusCode || 500;
+                    res.write(output);
+                    res.status(statusCode);
+                    res.end();
+                })
+                .catch(error => {
+                    next(error);
+                });
+            /**
+             * Dispatch END action so sagas stop listening after they're resolved
+             */
+            store.close();
+        } catch (e) {
+            next(e);
+        }
+    };
+}
+
+export function renderer(config: Config) {
+    return (req: Request, res: Response, next: NextFunction) => {
+        const staticContext: StaticContext = {
+            statusCode: 200,
+        };
+        try {
+            const store = configureStore("server");
+            const routes = Routing.getConfig(routeConfig, config.publicPath);
+            const matchedRoutes = Routing.getMatches(routes, {
+                as: "switch",
+                path: req.path,
+            });
+
+            /**
+             * Setup store to render application after sagas have complete
+             * @todo fetch data doesn't seem to be called
+             */
+            store
+                .runSaga(rootSaga)
+                .toPromise()
+                .then(() => {
+                    const output = Template({
+                        req,
+                        routes,
+                        staticContext,
+                        store,
+                    });
+                    const statusCode = staticContext.statusCode || 200;
+                    res.write(output);
+                    res.status(statusCode);
+                    res.end();
+                })
+                .catch(error => {
+                    next(error);
+                });
+            matchedRoutes.forEach(matchedRoute => {
+                if (matchedRoute.route.fetchData) {
+                    matchedRoute.route.fetchData(
+                        store.dispatch,
```

Comment:
What should this be changed to since `store` no longer exists?

Reply:
@ductiletoaster I deleted it but not sure what Dispatch to put in its place.

Reply:
So yeah the whole fetch data thing should be deleted. I think we should leave those details to the team making an app. 


I also want to try something completely different for server-side data retrieval using hooks